subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -683,7 +683,12 @@ static int mod_pub_status(struct bt_mesh_model *model,
 		}
 
 		if (param->pub) {
-			*param->pub = pub;
+			param->pub->addr = pub.addr;
```

Comment:
Or just set `pub.uuid` to `NULL` after line 659? Though here I'm uncertain if it is safer.

Reply:
What if uuid has been already initialized before calling mesh API? bsim test does it here: https://github.com/nrfconnect/sdk-zephyr/blob/main/tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c#L536
after that it initializes gotten publication being sure uuid is NULL: https://github.com/nrfconnect/sdk-zephyr/blob/main/tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c#L573-L575
API description doesn't claim it is prohibited. Even more uuid pointer is not the part of config server publication frame.

Reply:
Ok, fair enough.